.github/workflows/desktop-build.yml

```diff
@@ -235,6 +235,7 @@ jobs:
     runs-on: ${{matrix.os}}
     continue-on-error: ${{matrix.allow-failure == 'yes'}}
     env:
+      CCACHE_DIR: /tmp/${{matrix.os}}-${{matrix.type}}-cache
```

Comment:
reference: https://ccache.dev/manual/latest.html#_command_line_options 

ideally the Linux job should be updated too, but I didn't want to expand the scope of this PR

Reply:
I figured that the interaction with compile.sh and docker.sh scripts makes the setup a bit more complex and one needs to wrap his head around at first.
@ebbit1q's idea with the script was probably to have a simple and clean workflow configuration file for CI, and do the magic in the scripts.

There is some other cleanup in https://github.com/Cockatrice/Cockatrice/pull/5959 btw.

I also saw that we have Ninja now hardcoded in the docker.sh script. Ideally that should react on the configuration we have in the workflow file as well?